plugin-maven/src/main/java/com/diffplug/spotless/maven/SpotlessApplyMojo.java

```diff
@@ -33,25 +33,39 @@ public class SpotlessApplyMojo extends AbstractSpotlessMojo {
 
 	@Override
 	protected void process(Iterable<File> files, Formatter formatter, UpToDateChecker upToDateChecker) throws MojoExecutionException {
+		ImpactedFilesTracker impactedFilesTracker = new ImpactedFilesTracker();
+
 		for (File file : files) {
 			if (upToDateChecker.isUpToDate(file.toPath())) {
+				impactedFilesTracker.skippedAsCleanCache();
 				if (getLog().isDebugEnabled()) {
 					getLog().debug("Spotless will not format an up-to-date file: " + file);
 				}
 				continue;
 			}
 
 			try {
+				impactedFilesTracker.checked();
 				PaddedCell.DirtyState dirtyState = PaddedCell.calculateDirtyState(formatter, file);
 				if (!dirtyState.isClean() && !dirtyState.didNotConverge()) {
+					getLog().info(String.format("Writing clean file: %s", file));
 					dirtyState.writeCanonicalTo(file);
 					buildContext.refresh(file);
+					impactedFilesTracker.cleaned();
 				}
 			} catch (IOException e) {
 				throw new MojoExecutionException("Unable to format file " + file, e);
 			}
 
 			upToDateChecker.setUpToDate(file.toPath());
 		}
+
+		// We print the number of considered files which is useful when ratchetFrom is setup
+		int nbSkipped = impactedFilesTracker.getSkipped();
+		int nbChecked = impactedFilesTracker.getChecked();
+		int nbCleaned = impactedFilesTracker.getCleaned();
+		int totalProcessed = nbSkipped + nbChecked + nbCleaned;
```

Comment:
```
int totalProcessed = nbSkipped + nbChecked
int numAlreadyClean = nbChecked - nbCleaned
```

An alternative which might be clearer is to change the fields in `ImpactedFilesTracker`, e.g.

```java
if (upToDateChecker.isUpToDate(file.toPath())) {
  impactedFilesTracker.skippedAsCleanCache();
}
PaddedCell.DirtyState dirtyState = PaddedCell.calculateDirtyState(formatter, file);
if (!dirtyState.isClean() && !dirtyState.didNotConverge()) {
  impactedFilesTracker.cleaned()
} else {
  impactedFilesTracker.checkedButAlreadyClean()
}
```

With this formulation, `totalProcessed = skippedAsCleanCache + cleaned + checkedButAlreadyClean`

Reply:
I went for the second suggestion. It was the initial spirit (while the execution was... lacking rigour).